### ISSUE TITLE
Feature On Checking Connection Status Of Device

### ISSUE THREAD

**Issue by xjz** [thingsboard/thingsboard#642]
The feature for checking connection status of device seems still missing,there are some needs related to this in our use case:
1. a dashboard showing the number of online devices for a given user.
2. online or offline indicator for a device on web page.
3. push event to another system via POST when device goes online and offline.

This seems not to be an easy task,any suggestions on implementation?

**Comment by nucflash** [comment]
It seems that the MQTT protocol can support this type of functionality "out of the box" through Last Will Testament–unfortunately it is dependent on the actual implementation, and from a look I had in the code, ThingsBoard has its own implementation of MQTT and LWT isn't supported, yet.

If LWT were supported, it would be possible to do as easy as this:
```
...
client.will_set('v1/devices/me/attributes', {'online': true}, retain=False, qos=0)
client.connect()
client.publish('v1/devices/me/attributes', {'online': false})
...
```

Here's some links on the subject:
* https://stackoverflow.com/questions/17270863/mqtt-what-is-the-purpose-or-usage-of-last-will-testament
* https://www.hivemq.com/blog/mqtt-essentials-part-9-last-will-and-testament
* https://github.com/thingsboard/thingsboard/blob/8a28cd703fa63f9dbd12464088e6d25593e5aa86/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/MqttTransportHandler.java

**Comment by xjz** [comment]
it seems that will be included in v1.5
